user/caribou/module/src/AD9249EventConverter.cc

```diff
@@ -65,20 +74,44 @@ void AD9249Event2StdEventConverter::decodeChannel(
   timestamp = static_cast<uint64_t>(timestamp * 1e6 / 65.);
 }
 
-bool AD9249Event2StdEventConverter::Converting(
-    eudaq::EventSPC d1, eudaq::StandardEventSP d2,
-    eudaq::ConfigurationSPC conf) const {
+bool
+AD9249Event2StdEventConverter::Converting(eudaq::EventSPC d1,
+                                          eudaq::StandardEventSP d2,
+                                          eudaq::ConfigurationSPC conf) const {
 
-  if( !m_configured ){
+  if (!m_configured) {
     threshold_trig = conf->Get("threshold_trig", 1000);
     threshold_low = conf->Get("threshold_low", 101);
+    m_blStart = conf->Get("blStart", 200);
+    m_blEnd = conf->Get("blEnd", 100);
+    m_calib_range_min = conf->Get("calib_range_min", m_calib_range_min);
+    m_calib_range_max = conf->Get("calib_range_max", m_calib_range_max);
     m_useTime = conf->Get("use_time_stamp", false);
     m_waveform_filename = conf->Get("waveform_filename", "");
 
-    EUDAQ_DEBUG( "Using configuration:" );
-    EUDAQ_DEBUG( " threshold_low  = " + to_string( threshold_low ));
-    EUDAQ_DEBUG( " threshold_trig  = " + to_string( threshold_trig ));
-    EUDAQ_DEBUG( " use_time_stamp  = " + to_string( m_useTime ));
+    // read calibration functions
+    delete m_calib_functions.at(
+        0); // since there is only one 'name' this destroys them all
```

Comment:
Oh wow, that is quite a nice example of very sketchy memory management :/

Is there a reason you really need them to be pointers? Otherwise

```cpp
std::vector<TF1>
AD9249Event2StdEventConverter::m_calib_functions{};

m_calib_functions.clear();
for (unsigned int i = 0; i < m_calib_strings.size(); i++) {
    m_calib_functions.emplace_back(name.c_str(), m_calib_strings.at(i).c_str(), m_calib_range_min, _calib_range_max);
}
```

would be much cleaner and also more safe (right now you loop over `m_calib_strings` and use their index to access elements in `m_calib_functions` - this will fail (with `std::out_of_bounds`) if `m_calib_strings.size() != m_calib_functions.size()`.

Reply:
Yup, this is the better solution!